src/backend/UserDb.js

```diff
@@ -11,17 +11,31 @@ class UserDb {
         return UserDb.instance;
     }
 
-    async userExists(un, pw) {
-        const doc = await this.programDb.cfind({ username: un, password: pw })
-            .exec();
+    async userExists(username, password = null) {
+        let doc = [];
+        if (password !== null) {
+            doc = await this.programDb.cfind({ username: username, password: password }).exec();
+        } else {
+            doc = await this.programDb.cfind({ username: username}).exec();
+        }
         return doc.length !== 0;
     }
 
-    async addUser(un, pw) {
+    async addUser(username, password, role = "student") {
         // Needs to be updated for check but this is for testing
-        const doc = await this.programDb.insert({ username: un, password: pw });
+        const doc = await this.programDb.insert({ username: username, password: password, role: role });
         return doc.length !== 0;
     }
+
+    async updateRole(username, role = "student") {
+        let doc = [];
+        if (this.userExists(username)) {
+            doc = await this.programDb.update({ username: username, role: role});
+            return doc.length !== 0;
```

Comment:
Checking the length of the doc shouldn't be necessary, the doc returned is just the one you just inserted. Go ahead and return true to show it was successful